nltk/twitter/api.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# Natural Language Toolkit: Twitter API
+#
+# Copyright (C) 2001-2015 NLTK Project
+# Author: Ewan Klein <ewan@inf.ed.ac.uk>
+#         Lorenzo Rubio <lrnzcig@gmail.com>
+# URL: <http://nltk.org/>
+# For license information, see LICENSE.TXT
+
+"""
+This module provides an interface for TweetHandlers, and support for timezone
+handling.
+"""
+
+from datetime import tzinfo, timedelta, datetime
+import time as _time
+
+
+class LocalTimezoneOffsetWithUTC(tzinfo):
+    """
+    This is not intended to be a general purpose class for dealing with the
+    local timezone. In particular:
+
+    * it assumes that the date passed has been created using
+      `datetime(..., tzinfo=Local)`, where `Local` is an instance of
+      the object `LocalTimezoneOffsetWithUTC`;
+    * for such an object, it returns the offset with UTC, used for date comparisons.
+
+    Reference: https://docs.python.org/3/library/datetime.html
+    """
+    STDOFFSET = timedelta(seconds=-_time.timezone)
+
+    if _time.daylight:
+        DSTOFFSET = timedelta(seconds=-_time.altzone)
+    else:
+        DSTOFFSET = STDOFFSET
+
+    def utcoffset(self, dt):
+        """
+        Access the relevant time offset.
+        """
+        return self.DSTOFFSET
+
+LOCAL = LocalTimezoneOffsetWithUTC()
+
+class TweetHandlerI(object):
+    """
+    Interface class whose subclasses should implement a handle method that
+    Twitter clients can delegate to.
+    """
+    def __init__(self, limit=20, date_limit=None):
+        """
+        :param int limit: The number of data items to process in the current round of\
+        processing.
+
+        :param tuple date_limit: The date at which to stop collecting new\
+        data. This should be entered as a tuple which can serve as the\
+        argument to `datetime.datetime`. E.g. `data_limit=(2015, 4, 1, 12,\
+        40)` for 12:30 pm on April 1 2015.
+
+        """
+        self.limit = limit
+        self.date_limit = date_limit
+        if date_limit is not None:
+            self.date_limit = datetime(*date_limit, tzinfo=LOCAL)
+
+        self.startingup = True
+        """A flag to indicate whether this is the first data
+        item to be processed in the current round of processing."""
+        self.counter = 0
+
+    def handle(self, data):
+        """
+        Deal appropriately with data returned by the Twitter API
+        """
+        raise NotImplementedError
+
+    def handle_chunk(self, data_chunk):
+        """
+        Deal appropriately with a list of elements returned by the Twitter API
+        (default implementation should be enough in most cases)
```

Comment:
What does the return value mean? `handle_chunk` is not, to me, a boolean preposition.